lib/synapse/haproxy.rb

```diff
@@ -1030,19 +1030,33 @@ def generate_backend_stanza(watcher, config)
       ]
     end
 
+    def haproxy_exec(command)
+      info = nil
+      s = UNIXSocket.new(@opts['socket_file_path'])
+      s.write(command)
+      info = s.read
+    rescue StandardError => e
```

Comment:
i'm not sure about this `rescue` clause. i think we should allow the caller to decide how to handle exceptions, even though it does make the code more verbose at the call site. i can definitely envision cases where i want to query haproxy via the stats socket but *not* want to restart haproxy if that fails